turbopack/crates/turbo-tasks-backend/src/kv_backing_storage.rs

```diff
@@ -573,27 +641,38 @@ where
                     if let Some(batch) = batch {
                         let key = IntKey::new(*task_id);
                         let key = key.as_ref();
-                        if let Some(meta) = meta {
+                        if let Some((mut meta, rcstr_map)) = meta {
+                            let global_ids = save_strings_concurrent(batch, &rcstr_map)?;
+                            let mut meta_bytes = SmallVec::<[u8; 16]>::new();
+                            global_ids.write_to(&mut meta_bytes)?;
+                            meta_bytes.append(&mut meta);
```

Comment:
This is not very efficient, basically cloning the value everytime.

Maybe you want to put the global ids after the serialized bytes instead (putting the length in the last 4 bytes), so you can just append it to the buffer.
That's also not perfect, but at least it can use existing capacity that was left-over from serialization